app/controllers/favorite_tools_controller.rb

```diff
@@ -0,0 +1,40 @@
+class FavoriteToolsController < WithUserController
+  before_action :validate_ids
+
+  def index
+    render json: current_user_favorite_tools_response_json
+  end
+
+  def create
+    tool_ids.each do |tool_id|
+      current_user.favorite_tools.where(tool_id: tool_id).first_or_create
+    end
+    render json: current_user_favorite_tools_response_json
+  end
+
+  def destroy
+    current_user.favorite_tools.where(tool_id: tool_ids).delete_all
+    render json: current_user_favorite_tools_response_json
+  end
+
+  protected
+
+  def tool_ids
+    return [] unless params[:data] && params[:data].is_a?(Array)
+    return @tool_ids if @tool_ids
+
+    @tool_ids = params[:data].collect{ |data|
+      data[:id]
+    }
+  end
+
+  def validate_ids
+    missing = tool_ids - Resource.pluck(:id).collect(&:to_s)
+    error_response = missing.collect{ |tool_id| [tool_id, "invalid tool id"] }
+    render(json: { errors: error_response }) if error_response.any?
+  end
+
+  def current_user_favorite_tools_response_json
+    { "data" => current_user.favorite_tools.collect { |fav| { "type" => "resource", "id" => fav.tool_id.to_s } } }
```

Comment:
does this not work using the standard jsonapi serializer?

Reply:
Perhaps something about the requirements specified this JSON? (static "resource", no user_id)

Reply:
so, the requirements did include the sample jsonapi. the list of favorite tools is specifically a jsonapi resource identifier object pointing at a tool (`resource` within the mobile-content-api).

I was kind of expecting it to be seemless json generation like most of the rest of the API. But this is also the first time we've done a relationship only API endpoint like this

Reply:
@knutsenm in case you were curious, this is the relevant part of the jsonapi spec for this API: https://jsonapi.org/format/#crud-updating-to-many-relationships

Reply:
ah, I think it's not relying on the default serializer because it's using the `favorite_tools` relationship directly instead of resolving it to the actual `resource` objects.

Reply:
I'm fine with this logic persisting as is because it appears to allow a DB optimization.